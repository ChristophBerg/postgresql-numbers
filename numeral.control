default_version = '1'
comment = 'numeral datatypes extension'
relocatable = true
