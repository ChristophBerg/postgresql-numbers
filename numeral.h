/* type defs */

typedef long long Numeral;
typedef long long Zahl;
typedef long long Roman;

/* prototypes */

const char *numeral_cstring (Numeral numeral);
int numeral_parse (char *s, Numeral *numeral);

const char *zahl_cstring (Zahl zahl);
int zahl_parse (char *s, Zahl *zahl);

const char *roman_cstring (Roman roman);
int roman_parse (char *s, Roman *roman);
