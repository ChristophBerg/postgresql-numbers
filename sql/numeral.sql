SELECT '0'::numeral;
SELECT '1'::numeral;
SELECT '-2'::numeral;

SELECT 'zero'::numeral;

SELECT 'one'::numeral;
SELECT 'minus one'::numeral;
SELECT 'two'::numeral;
SELECT 'three'::numeral;
SELECT 'four'::numeral;
SELECT 'five'::numeral;
SELECT 'six'::numeral;
SELECT 'seven'::numeral;
SELECT 'eight'::numeral;
SELECT 'nine'::numeral;
SELECT 'ten'::numeral;

SELECT 'eleven'::numeral;
SELECT 'twelve'::numeral;
SELECT 'thirteen'::numeral;
SELECT 'fourteen'::numeral;
SELECT 'fifteen'::numeral;
SELECT 'sixteen'::numeral;
SELECT 'seventeen'::numeral;
SELECT 'eighteen'::numeral;
SELECT 'nineteen'::numeral;

SELECT 'twenty'::numeral;
SELECT 'twenty-one'::numeral;
SELECT 'thirty'::numeral;
SELECT 'thirty-two'::numeral;
SELECT 'forty'::numeral;
SELECT 'fifty'::numeral;
SELECT 'sixty'::numeral;
SELECT 'seventy'::numeral;
SELECT 'eighty'::numeral;
SELECT 'ninety'::numeral;

SELECT 'hundred'::numeral;
SELECT 'one hundred'::numeral;
SELECT 'hundred one'::numeral;
SELECT 'one hundred one'::numeral;
SELECT 'one hundred twenty-one'::numeral;

SELECT 'two hundred'::numeral;
SELECT 'two hundred one'::numeral;
SELECT 'two hundred thirty-two'::numeral;

SELECT 'thousand'::numeral;
SELECT 'one thousand'::numeral;
SELECT 'one thousand one'::numeral;
SELECT 'thousand one hundred'::numeral;
SELECT 'one thousand one hundred'::numeral;
SELECT 'eleven hundred'::numeral;
SELECT 'one thousand one hundred one'::numeral;

SELECT 'hundred thousand'::numeral;
SELECT 'one hundred thousand'::numeral;
SELECT 'one hundred one thousand'::numeral;
SELECT 'two hundred thousand'::numeral;

SELECT 'one million'::numeral;
SELECT 'one million one'::numeral;
SELECT 'one million one hundred'::numeral;
SELECT 'one million one thousand'::numeral;
SELECT 'one million one hundred thousand'::numeral;
SELECT 'two million'::numeral;

SELECT 'hundred million'::numeral;

SELECT 'one billion'::numeral;
SELECT 'one billion one million'::numeral;
SELECT 'one billion one million one thousand'::numeral;

SELECT 'billion'::numeral;
SELECT 'trillion'::numeral;
SELECT 'quadrillion'::numeral;
SELECT 'quintillion'::numeral;
