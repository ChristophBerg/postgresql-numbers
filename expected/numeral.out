SELECT '0'::numeral;
 numeral 
---------
 zero
(1 row)

SELECT '1'::numeral;
 numeral 
---------
 one
(1 row)

SELECT '-2'::numeral;
  numeral  
-----------
 minus two
(1 row)

SELECT 'zero'::numeral;
 numeral 
---------
 zero
(1 row)

SELECT 'one'::numeral;
 numeral 
---------
 one
(1 row)

SELECT 'minus one'::numeral;
  numeral  
-----------
 minus one
(1 row)

SELECT 'two'::numeral;
 numeral 
---------
 two
(1 row)

SELECT 'three'::numeral;
 numeral 
---------
 three
(1 row)

SELECT 'four'::numeral;
 numeral 
---------
 four
(1 row)

SELECT 'five'::numeral;
 numeral 
---------
 five
(1 row)

SELECT 'six'::numeral;
 numeral 
---------
 six
(1 row)

SELECT 'seven'::numeral;
 numeral 
---------
 seven
(1 row)

SELECT 'eight'::numeral;
 numeral 
---------
 eight
(1 row)

SELECT 'nine'::numeral;
 numeral 
---------
 nine
(1 row)

SELECT 'ten'::numeral;
 numeral 
---------
 ten
(1 row)

SELECT 'eleven'::numeral;
 numeral 
---------
 eleven
(1 row)

SELECT 'twelve'::numeral;
 numeral 
---------
 twelve
(1 row)

SELECT 'thirteen'::numeral;
 numeral  
----------
 thirteen
(1 row)

SELECT 'fourteen'::numeral;
 numeral  
----------
 fourteen
(1 row)

SELECT 'fifteen'::numeral;
 numeral 
---------
 fifteen
(1 row)

SELECT 'sixteen'::numeral;
 numeral 
---------
 sixteen
(1 row)

SELECT 'seventeen'::numeral;
  numeral  
-----------
 seventeen
(1 row)

SELECT 'eighteen'::numeral;
 numeral  
----------
 eighteen
(1 row)

SELECT 'nineteen'::numeral;
 numeral  
----------
 nineteen
(1 row)

SELECT 'twenty'::numeral;
 numeral 
---------
 twenty
(1 row)

SELECT 'twenty-one'::numeral;
  numeral   
------------
 twenty-one
(1 row)

SELECT 'thirty'::numeral;
 numeral 
---------
 thirty
(1 row)

SELECT 'thirty-two'::numeral;
  numeral   
------------
 thirty-two
(1 row)

SELECT 'forty'::numeral;
 numeral 
---------
 forty
(1 row)

SELECT 'fifty'::numeral;
 numeral 
---------
 fifty
(1 row)

SELECT 'sixty'::numeral;
 numeral 
---------
 sixty
(1 row)

SELECT 'seventy'::numeral;
 numeral 
---------
 seventy
(1 row)

SELECT 'eighty'::numeral;
 numeral 
---------
 eighty
(1 row)

SELECT 'ninety'::numeral;
 numeral 
---------
 ninety
(1 row)

SELECT 'hundred'::numeral;
   numeral   
-------------
 one hundred
(1 row)

SELECT 'one hundred'::numeral;
   numeral   
-------------
 one hundred
(1 row)

SELECT 'hundred one'::numeral;
     numeral     
-----------------
 one hundred one
(1 row)

SELECT 'one hundred one'::numeral;
     numeral     
-----------------
 one hundred one
(1 row)

SELECT 'one hundred twenty-one'::numeral;
        numeral         
------------------------
 one hundred twenty-one
(1 row)

SELECT 'two hundred'::numeral;
   numeral   
-------------
 two hundred
(1 row)

SELECT 'two hundred one'::numeral;
     numeral     
-----------------
 two hundred one
(1 row)

SELECT 'two hundred thirty-two'::numeral;
        numeral         
------------------------
 two hundred thirty-two
(1 row)

SELECT 'thousand'::numeral;
   numeral    
--------------
 one thousand
(1 row)

SELECT 'one thousand'::numeral;
   numeral    
--------------
 one thousand
(1 row)

SELECT 'one thousand one'::numeral;
     numeral      
------------------
 one thousand one
(1 row)

SELECT 'thousand one hundred'::numeral;
         numeral          
--------------------------
 one thousand one hundred
(1 row)

SELECT 'one thousand one hundred'::numeral;
         numeral          
--------------------------
 one thousand one hundred
(1 row)

SELECT 'eleven hundred'::numeral;
         numeral          
--------------------------
 one thousand one hundred
(1 row)

SELECT 'one thousand one hundred one'::numeral;
           numeral            
------------------------------
 one thousand one hundred one
(1 row)

SELECT 'hundred thousand'::numeral;
       numeral        
----------------------
 one hundred thousand
(1 row)

SELECT 'one hundred thousand'::numeral;
       numeral        
----------------------
 one hundred thousand
(1 row)

SELECT 'one hundred one thousand'::numeral;
         numeral          
--------------------------
 one hundred one thousand
(1 row)

SELECT 'two hundred thousand'::numeral;
       numeral        
----------------------
 two hundred thousand
(1 row)

SELECT 'one million'::numeral;
   numeral   
-------------
 one million
(1 row)

SELECT 'one million one'::numeral;
     numeral     
-----------------
 one million one
(1 row)

SELECT 'one million one hundred'::numeral;
         numeral         
-------------------------
 one million one hundred
(1 row)

SELECT 'one million one thousand'::numeral;
         numeral          
--------------------------
 one million one thousand
(1 row)

SELECT 'one million one hundred thousand'::numeral;
             numeral              
----------------------------------
 one million one hundred thousand
(1 row)

SELECT 'two million'::numeral;
   numeral   
-------------
 two million
(1 row)

SELECT 'hundred million'::numeral;
       numeral       
---------------------
 one hundred million
(1 row)

SELECT 'one billion'::numeral;
   numeral   
-------------
 one billion
(1 row)

SELECT 'one billion one million'::numeral;
         numeral         
-------------------------
 one billion one million
(1 row)

SELECT 'one billion one million one thousand'::numeral;
               numeral                
--------------------------------------
 one billion one million one thousand
(1 row)

SELECT 'billion'::numeral;
   numeral   
-------------
 one billion
(1 row)

SELECT 'trillion'::numeral;
   numeral    
--------------
 one trillion
(1 row)

SELECT 'quadrillion'::numeral;
     numeral     
-----------------
 one quadrillion
(1 row)

SELECT 'quintillion'::numeral;
     numeral     
-----------------
 one quintillion
(1 row)

